tests/ccl_test_power_nu.c

```diff
@@ -0,0 +1,286 @@
+  
+  if (i_model==1){
+  
+      params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu0, data-> mnu_type, 
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  } else if (i_model==2){
+	  params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu1, data->mnu_type,
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  } else if (i_model==3){
+	 params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu2, data->mnu_type,
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  }
+
+  ccl_cosmology * cosmo_linear = ccl_cosmology_create(params, config_linear);
+  ASSERT_NOT_NULL(cosmo_linear);
+  
+  sprintf(fname,"./tests/benchmark/model%d_pk_nu.txt",i_model);
+  f=fopen(fname,"r");
+  if(f==NULL) {
+    fprintf(stderr,"Error opening file %s\n",fname);
+    exit(1);
+  }
+  nk=linecount(f)-1; rewind(f);
+  
+  double k=0.,pk_bench=0.,pk_ccl,err, k_h, pk_h;
+  double z=0.; //Other redshift checks are possible but not currently implemented
+  int stat=0;
+
+	for(i=0;i<nk;i++) {      
+    stat=fscanf(f,"%le %le\n",&k_h, &pk_h);
+    if(stat!=2) {
+      fprintf(stderr,"Error reading file %s, line %d\n",fname,i);
+      exit(1);
+    }
+    k=k_h*data->h;
+    pk_bench=pk_h/pow(data->h,3);
+    
+    pk_ccl=ccl_linear_matter_power(cosmo_linear,k,1./(1+z),&status);
+    if (status) printf("%s\n",cosmo_linear->status_message);
+    err=fabs(pk_ccl/pk_bench-1);
+    ASSERT_DBL_NEAR_TOL(err,0.,POWER_NU_TOL);
+    }
+    
+  fclose(f);
+
+  ccl_cosmology_free(cosmo_linear);
+
+}
+
+static void compare_power_nu_nl(int i_model,struct power_nu_nl_data * data)
+{
+  int i,j, nk_nl;
+  int status=0;
+  char fname_nl[256], str[1024];
+  FILE *f_nl;
+  ccl_configuration config_nonlinear = default_config;
+  config_nonlinear.matter_power_spectrum_method = ccl_halofit;
+  
+  ccl_parameters params;
+  
+  if (i_model==1){
+  
+      params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu0, data-> mnu_type, 
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  } else if (i_model==2){
+	  params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu1, data->mnu_type,
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  } else if (i_model==3){
+	 params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+						data->Neff, data->mnu2, data->mnu_type,
+						data->w_0[i_model-1], data->w_a[i_model-1],
+						data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
+  }
```

Comment:
These all look identical to me.  Why the three cases?

Reply:
They are slightly different, actually. See the data->mnu entry in each of them, they vary according to the model.

Reply:
Ah, in that case, I'd suggest writing this as:
```
params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k, 
                                                data->Neff, 
                                                i_model==1 ? data->mnu0 : i_model==2 ? data->mnu1 : data->mnu2,
                                                data->mnu_type,
						data->w_0[i_model-1], data->w_a[i_model-1],						 
                                                data->h, data->A_s, data->n_s,-1,-1,-1,-1,NULL,NULL, &status);
```
to make this more apparent.  Or even use a temporary variable to set the mnu value to be used for each case, rather than repeat the whole thing three times, just changing one character.